default.nix

```diff
@@ -38,6 +38,8 @@ let
 
     tests = collectComponents "tests" isCardanoWallet haskellPackages;
     benchmarks = collectComponents "benchmarks" isCardanoWallet haskellPackages;
+    checks = pkgs.recurseIntoAttrs (pkgs.lib.mapAttrs (_: package: package.checks)
```

Comment:
Needs a comment saying that `tests` is the test suites which have been built, and `checks` are the result of executing the tests.

The mapping code could be split into a util function I think.